File: V1-Java/app/src/main/java/com/kumoh/iat/CustomTableView.java
package com.kumoh.iat;

import android.content.Context;
import android.util.AttributeSet;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TableLayout;

public class CustomTableView extends TableLayout {

    Context context;

    public CustomTableView(Context context, AttributeSet attrs) {
        super(context, attrs);
        this.context = context;
    }

    public void setAdapter(ArrayAdapter adapter) {
        try {
            if (adapter == null) return;

            for (int i = 0; i < adapter.getCount(); i++) {
                addView(adapter.getView(i, null, this));
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: V2-Kotlin/app/src/main/java/kumoh/iat/v2/ui/customview/CustomTableView.kt
package kumoh.iat.v2.ui.customview

import android.content.Context
import android.util.AttributeSet
import android.widget.ArrayAdapter
import android.widget.TableLayout

class CustomTableView(
    context: Context,
    attrs: AttributeSet?
) : TableLayout(context, attrs) {

    fun setAdapter(adapter: ArrayAdapter<*>?) {
        if (adapter == null) return

        for (i in 0 until adapter.count) {
            addView(adapter.getView(i, null, this))
        }
    }
}
